ufl/mathfunctions.py

```diff
@@ -282,10 +321,14 @@ class Erf(MathFunction):
     __slots__ = ()
 
     def __new__(cls, argument):
-        if isinstance(argument, (ScalarValue, Zero)):
+        if isinstance(argument, (IntValue, FloatValue, Zero)):
```

Comment:
I don't like any of these splits of `ScalarValue` into `IntValue` and `FloatValue`. What if there would be other real scalars, e.g., `RationalValue`? I think it's better to leave these as `ScalarValue`, and have a nested if inside to check if it is in fact `ComplexValue`.